hyperspy/drawing/_widgets/polygon.py

```diff
@@ -0,0 +1,304 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2022 The HyperSpy developers
+#
+# This file is part of HyperSpy.
+#
+# HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HyperSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+from matplotlib.path import Path
+from matplotlib.widgets import PolygonSelector
+
+from hyperspy.drawing.widgets import MPLWidgetBase
+
+
+class PolygonWidget(MPLWidgetBase):
+
+    """PolygonWidget is a widget for drawing one or more arbitrary
+    polygons, which can then be used as a region-of-interest.
+    The active polygon can be moved by shift-clicking.
+    A polygon vertex can be moved by clicking its handle. If incomplete,
+    it is also necessary to press control.
+    The active polygon can be deleted by pressing escape.
+    To delete other polygons, click inside it until it has a red outline,
+    then press `Delete`.
+    """
+
+    def __init__(self, axes_manager, mpl_ax=None, polygons=None, **kwargs):
```

Comment:
To keep the API consistent across widgets, we should avoid adding other argument, for example, here: `mpl_ax` and `polygons`.

If there is a need for these arguments, it would be better to implement it in a separate PR for all widgets.

Reply:
Yes indeed, these should be removed here! They were added to help development, but should be removed now that there are more conforming ways of doing this.